papaparse.js

```diff
@@ -1693,17 +1696,37 @@ License: MIT
 			}
 
 			/**
-             * checks if there are extra spaces after closing quote and given index without any text
-             * if Yes, returns the number of spaces
-             */
+			 * Checks if there are insignificant whitespace characters after a closing quote and before
+			 * the next delimiter or newline. If so, returns the number of such characters.
+			 * This function has been optimized to avoid creating substrings for performance.
+			 */
 			function extraSpaces(index) {
+				// If index is -1, delimiter/newline wasn't found after the quote.
+				if (index === -1) {
+					return 0;
+				}
+
+				var start = quoteSearch + 1;
+				// Ensure start is not past the end of the input or the index
+				// Also check inputLen which is defined in the outer scope
+				if (start >= inputLen || start >= index) {
+					return 0;
+				}
+
 				var spaceLength = 0;
-				if (index !== -1) {
-					var textBetweenClosingQuoteAndIndex = input.substring(quoteSearch + 1, index);
-					if (textBetweenClosingQuoteAndIndex && textBetweenClosingQuoteAndIndex.trim() === '') {
-						spaceLength = textBetweenClosingQuoteAndIndex.length;
+				for (var i = start; i < index; i++) {
+					var charCode = input.charCodeAt(i);
+					// Check for space (32) or tab (9)
+					// This aims to match the common behavior implicitly handled by the previous .trim() check
```

Comment:
I do not think we should reimplement trim function but just call it. Please keep the previous trimg function. 

There reason is that trim may remove more empty chars that I do not want to maintain as part of the library.